### ISSUE TITLE
[red team] Attempt to escape github actions executor and upload an artefact

### ISSUE THREAD

**Issue by alsuren** [alsuren/cargo-quickinstall#49]

**Comment by alsuren** [comment]
I *think* that this is now solved. I will leave this "in review" until I can actually verify that it leaves me with a properly neutered token, and also that it's not possible to upload artifacts to workflow runs that are not the one your job is for.

Things to try would be:

- [ ] spin up a GitHub runner (https://github.com/actions/runner) inside a VM on a laptop, or in the cloud somewhere.
- [ ] Add a bunch of logging
- [ ] make a new repo with a job that just prints the token and sleeps forever, and assign it to that runner
- [ ] take the token and try to do stuff with it
- [ ] trigger another workflow run in parallel on a different runner (that also sleeps forever)
- [ ] try to use actions/upload-artifact to upload from one workflow run to the other

**Comment by alsuren** [comment]
As part of this ticket, I think that I would also like to go around and trim down the permissions used in all other jobs. The cronjob and upload job both currently use a deploy key that is a bit overpowered. It would be nice to have at least the upload job go back to using a less powerful key.
